movement-migration/validation-tool/src/types/storage.rs

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Context;
+use aptos_config::config::{StorageConfig, StorageDirPaths};
+use aptos_db::AptosDB;
+use aptos_storage_interface::{
+    state_store::state_view::db_state_view::{DbStateView, DbStateViewAtVersion},
+    DbReader,
+};
+use aptos_types::state_store::state_key::StateKey;
+use either::Either;
+use std::ops::Deref;
+use std::path::PathBuf;
+use std::sync::Arc;
+use tracing::debug;
+
+pub struct Storage(Arc<dyn DbReader>);
+
+impl Storage {
+    pub fn open(path: &PathBuf) -> anyhow::Result<Self> {
+        let config = StorageConfig::default();
+        let aptos_db = AptosDB::open(
```

Comment:
You use the same Aptos core version for both DB. We should use the old and new version of the framework and do the comparison. For example, if loading the old DB with the new framework change the ledger version, you won't see it. Both DB the old and the one to validate has been loaded with the same code and the ledger version change has been applied to both. We want to be sure we apply the state verification at the ledger version of the old db. It's the same for all state. 
Perhaps you've done some test to justify to use the same version, so a detailed comment should be put here to justify it.

Reply:
I'm opening the databases in read-only mode, so the framework cannot apply changes (for example, changing the ledger version). However, the framework is backward compatible, and the newest version of `AptosDB` should be able to open the old database.